src/libraries/Microsoft.Extensions.Http/tests/Microsoft.Extensions.Http.Tests/HttpClientFactoryDisposeTests.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.Extensions.Http.Tests
+{
+    public class HttpClientFactoryDisposeTests
+    {
+        [Fact]
+        public async Task DisposingServiceProvider_DisposesHttpClientFactory_ReleasesResources()
+        {
+            // Arrange
+            var disposeCounter = new DisposeCounter();
+            var services = new ServiceCollection();
+            services.AddSingleton(disposeCounter);
+            services.AddTransient<DisposeTrackingHandler>();
+
+            services.AddHttpClient("test")
+                .AddHttpMessageHandler<DisposeTrackingHandler>()
+                .SetHandlerLifetime(TimeSpan.FromMilliseconds(50)); // Very short to ensure quick expiration
+
+            var serviceProvider = services.BuildServiceProvider();
+            var factory = serviceProvider.GetRequiredService<IHttpClientFactory>();
+
+            // Act - create and use clients
+            for (int i = 0; i < 5; i++)
+            {
+                using var client = factory.CreateClient("test");
```

Comment:
if we use the same name in CreateClient, the factory will reuse the same handler. actually, the "minimal lifetime" assigned above also intoduces a race on top. so this test won't be stable, and it might have all the range from 1 to 5 handlers depending on how fast/slow the machine it runs on is.

I assume we need to use 5 different names here.

also, 5 should be a variable/constant, especially given that it is used in assert